### PR TITLE
Add warning about datamodel-code-generator to docs

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: docs/integrations/datamodel_code_generator.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Code Generation with datamodel-code-generator
 
+!!! warning
+    [datamodel-code-generator still does not support Pydantic V2](https://github.com/koxudaxi/datamodel-code-generator/issues/803).
```
Comment (Kludex) [inline docs/integrations/datamodel_code_generator.md:4]: Are we sure it doesn't work? I asked without confirming.

Comment (hramezani) [reply docs/integrations/datamodel_code_generator.md:4]: Not sure. but as there is an open issue for that and also its [dependency on Pydantic V1](https://github.com/koxudaxi/datamodel-code-generator/blob/82e45f4f5d1801ae95cbc87bf355561bdfc088de/pyproject.toml#L74C12-L74C19), I think it does not work.

Comment (Kludex) [reply docs/integrations/datamodel_code_generator.md:4]: It says "Likely requires" on the issue... 👀 🤷‍♂️

Comment (hramezani) [reply docs/integrations/datamodel_code_generator.md:4]: Yeah, but probably it requires a new release.
Do I need to change the warning 🤔 ?

**Review by Kludex — APPROVED** [review]

**Pull request merged**